### PR TITLE
8338469: com/sun/jdi/DataDumpTest.java failed with Not a debuggee, or not listening for debugger to attach

### PR DESCRIPTION
**Pull request by plummercj**
There are issues with the test attaching to the debuggee too soon, and the debug agent isn't ready. yet. This test is based on ProcessAttachTest, which does not have this issue. I eventually realized the reason why is because ProcessAttachTest has this little snippet of code, which I had removed from DataDumpTest:

             // Wait for the process to start
            InputStream is = p.getInputStream();
            is.read(); 

This is waiting for the start of the debug agent's "Listening..." message. I've re-added this to DataDumpTest, and it seems to fix the issue. Tested by running 20 times with `-Xcomp` on all supported platforms with no failures. It used to fail about 2/3 of the time.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338469](https://bugs.openjdk.org/browse/JDK-8338469): com/sun/jdi/DataDumpTest.java failed with Not a debuggee, or not listening for debugger to attach (**Bug** - P4)

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20604/head:pull/20604` \
`$ git checkout pull/20604`

Update a local copy of the PR: \
`$ git checkout pull/20604` \
`$ git pull https://git.openjdk.org/jdk.git pull/20604/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20604`

View PR using the GUI difftool: \
`$ git pr show -t 20604`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20604.diff">https://git.openjdk.org/jdk/pull/20604.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20604#issuecomment-2292357054)

### CODE REVIEW

**Comment by plummercj** [comment]
Ping! I'd like to get this pushed today since it is making a lot of noise in our CI.

File: test/jdk/com/sun/jdi/DataDumpTest.java [hunk 1]
```diff
@@ -93,7 +98,7 @@ private static void runTest(String jdwpArg) throws Exception {
             out.waitFor(); // Wait for the debuggee to exit
 
             System.out.println("Deuggee output:");
```
Comment (dcubed-ojdk) [inline test/jdk/com/sun/jdi/DataDumpTest.java:100]: nit typo: s/Deuggee/Debuggee/

Not your typo, but could you fix it while you are there?

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. Thanks for including `firstChar` in the debuggee output.
I think this is a trivial fix.

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. Thanks for fixing the typo.

This is a trivial fix.

**Comment by plummercj** [comment]
Thanks Dan!

/integrate

**Pull request closed without merging**